notebooks/demo/nx_cugraph_demo.ipynb

```diff
@@ -0,0 +1,672 @@
+   "source": [
+    "import networkx as nx\n",
+    "karate_club_graph = nx.karate_club_graph()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Having `networkx` compute the `betweenness_centrality` values for each node on this graph is quick and easy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2.51 ms ± 1.02 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%timeit global karate_nx_bc_results\n",
+    "karate_nx_bc_results = nx.betweenness_centrality(karate_club_graph)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Automatic GPU acceleration\n",
+    "When `nx-cugraph` is installed, `networkx` will detect it on import and make it available as a backend for APIs supported by that backend.  However, `networkx` does not assume the user always wants to use a particular backend, and instead looks at various configuration mechanisms in place for users to specify how `networkx` should use installed backends. Since `networkx` was not configured to use a backend for the above `betweenness_centrality` call, it used the default implementation provided by NetworkX.\n",
+    "\n",
+    "The first configuration mechanism to be demonstrated below is the `NETWORKX_AUTOMATIC_BACKENDS` environment variable.  This environment variable directs `networkx` to use the backend specified everywhere it's supported, and does not require the user to modify any of their existing NetworkX code.\n",
+    "\n",
+    "To use it, a user sets `NETWORKX_AUTOMATIC_BACKENDS` in their shell to the backend they'd like to use.  If a user has more than one backend installed, the environment variable can also accept a comma-separated list of backends, ordered by priority in which `networkx` should use them, where the first backend that supports a particular API call will be used.  For example:\n",
+    "```\n",
+    "bash> export NETWORKX_AUTOMATIC_BACKENDS=cugraph\n",
+    "bash> python my_nx_app.py  # uses nx-cugraph wherever possible, then falls back to default implementation where it's not.\n",
+    "```\n",
+    "or in the case of multiple backends installed\n",
+    "```\n",
+    "bash> export NETWORKX_AUTOMATIC_BACKENDS=cugraph,graphblas\n",
+    "bash> python my_nx_app.py  # uses nx-cugraph if possible, then nx-graphblas if possible, then default implementation.\n",
+    "```\n",
+    "\n",
+    "`networkx` looks at the environment variable and the installed backends at import time, and will not re-examine the environment after that.  Because `networkx` was already imported in this notebook, the `reimport_nx()` utility will be called after the `os.environ` disctionary is updated to simulate an environment variable being set in the shell.\n",
```

Comment:
```suggestion
    "`networkx` looks at the environment variable and the installed backends at import time and will not re-examine the environment after that.  Because `networkx` was already imported in this notebook, the `reimport_nx()` utility will be called after the `os.environ` dictionary is updated to simulate an environment variable being set in the shell.\n",
```